rcl_action/test/rcl_action/test_action_interaction.cpp

```diff
@@ -0,0 +1,906 @@
+  ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  // Wait for result response to be ready
+  ret = rcl_wait(&this->wait_set, RCL_S_TO_NS(1));
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  ret = rcl_action_client_wait_set_get_entities_ready(
+    &this->wait_set,
+    &this->action_client,
+    &this->is_feedback_ready,
+    &this->is_status_ready,
+    &this->is_goal_response_ready,
+    &this->is_cancel_response_ready,
+    &this->is_result_response_ready);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  EXPECT_TRUE(this->is_result_response_ready);
+  EXPECT_FALSE(this->is_cancel_response_ready);
+  EXPECT_FALSE(this->is_feedback_ready);
+  EXPECT_FALSE(this->is_status_ready);
+  EXPECT_FALSE(this->is_goal_response_ready);
+
+  // Take result response with valid arguments
+  ret = rcl_action_take_result_response(
+    &this->action_client, &request_header, &incoming_result_response);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  // Check that the result response was received correctly
+  EXPECT_EQ(outgoing_result_response.status, incoming_result_response.status);
+  ASSERT_EQ(outgoing_result_response.sequence.size, incoming_result_response.sequence.size);
+  EXPECT_TRUE(!memcmp(
+      outgoing_result_response.sequence.data,
+      incoming_result_response.sequence.data,
+      outgoing_result_response.sequence.size));
+
+  test_msgs__action__Fibonacci_Goal_Request__fini(&outgoing_goal_request);
+  test_msgs__action__Fibonacci_Goal_Request__fini(&incoming_goal_request);
+  test_msgs__action__Fibonacci_Goal_Response__fini(&incoming_goal_response);
+  test_msgs__action__Fibonacci_Goal_Response__fini(&outgoing_goal_response);
+  test_msgs__action__Fibonacci_Result_Request__fini(&incoming_result_request);
+  test_msgs__action__Fibonacci_Result_Request__fini(&outgoing_result_request);
+  test_msgs__action__Fibonacci_Result_Response__fini(&incoming_result_response);
+  test_msgs__action__Fibonacci_Result_Response__fini(&outgoing_result_response);
+}
+
+
+// Exercises the "Example 2" sequence diagram found in the actions_proposal document.
+// This example is almost identical to the first, but this time the action client requests
+// for the goal to be canceled mid-execution. Note that the user defined method is allowed
+// to perform any shutdown operations after the cancel request before returning with the
+// cancellation result.
+TEST_F(
+  CLASSNAME(TestActionClientServerInteraction, RMW_IMPLEMENTATION), test_interaction_with_cancel)
+{
+  action_msgs__srv__CancelGoal_Request outgoing_cancel_request;
+  action_msgs__srv__CancelGoal_Request incoming_cancel_request;
+  action_msgs__srv__CancelGoal_Response outgoing_cancel_response;
+  action_msgs__srv__CancelGoal_Response incoming_cancel_response;
+  test_msgs__action__Fibonacci_Goal_Request outgoing_goal_request;
```

Comment:
Since the goal service, result service, and feedback messages are used in both tests, we can move them to the test fixutre and the init/fini calls to the SetUp/TearDown. The Cancel-specific messages can remain here.

Reply:
Just did as suggested, thanks!